lib/MemoryClient.js

```diff
@@ -1,51 +1,58 @@
+let _Promise
 export default class MemoryClient {
   constructor (options) {
     this.options = options
     this.data = {}
+    _Promise = options.promiseDependency || global.Promise
   }
 
   get (key, options, cb) {
     if (typeof options === 'function') {
       cb = options
     }
-
-    cb(null, this.data[key] || null)
+    return new _Promise((resolve, reject) => {
+      cb = cb || ((err, result) => (err ? reject(err) : resolve(result)))
+      cb(null, this.data[key] || null)
```

Comment:
You need to resolve or the promise will get 'stuck' and consume memory.

Reply:
I think this should now be resolved, it'll call resolve on the promise and call the callback.